frontend/src/metabase/actions/containers/ActionCreator/InlineActionSettings.tsx

```diff
@@ -0,0 +1,146 @@
+import React from "react";
+import { t } from "ttag";
+import _ from "underscore";
+import { connect } from "react-redux";
+
+import * as Urls from "metabase/lib/urls";
+import { getSetting } from "metabase/selectors/settings";
+
+import type { WritebackAction } from "metabase-types/api";
+import type { State } from "metabase-types/store";
+
+import Tooltip from "metabase/core/components/Tooltip";
+import Button from "metabase/core/components/Button";
+import Toggle from "metabase/core/components/Toggle";
+import SidebarContent from "metabase/query_builder/components/SidebarContent";
+import { useUniqueId } from "metabase/hooks/use-unique-id";
+
+import Icon from "metabase/components/Icon";
+import Actions from "metabase/entities/actions/actions";
+import ConfirmContent from "metabase/components/ConfirmContent";
+import Modal from "metabase/components/Modal";
+import { useToggle } from "metabase/hooks/use-toggle";
+import CopyWidget from "metabase/components/CopyWidget";
+
+import {
+  ActionSettingsContainer,
+  ActionSettingsContent,
+  CopyWidgetContainer,
+  ToggleContainer,
+  ToggleLabel,
+} from "./InlineActionSettings.styled";
+
+type PublicWritebackAction = WritebackAction & {
+  public_uuid: string;
+};
+
+interface OwnProps {
+  onClose: () => void;
+  actionId: number;
+}
+
+interface EntityLoaderProps {
+  action: WritebackAction;
+}
+
+interface StateProps {
+  siteUrl: string;
+  createPublicLink: ({ id }: { id: number }) => void;
+  deletePublicLink: ({ id }: { id: number }) => void;
+}
+
+type ActionSettingsInlineProps = OwnProps & EntityLoaderProps & StateProps;
+
+export const ActionSettingsTriggerButton = ({
+  onClick,
+}: {
+  onClick: () => void;
+}) => (
+  <Tooltip tooltip={t`Action settings`}>
+    <Button onlyIcon onClick={onClick} icon="gear" iconSize={16} />
+  </Tooltip>
+);
+
+const InlineActionSettings = ({
+  onClose,
+  actionId,
+  action,
+  siteUrl,
+  createPublicLink,
+  deletePublicLink,
+}: ActionSettingsInlineProps) => {
+  const id = useUniqueId();
+  const isPublic = isActionPublic(action);
+
+  const [isModalOpen, { turnOn: openModal, turnOff: closeModal }] = useToggle();
+
+  const handleTogglePublic = (isPublic: boolean) => {
+    if (isPublic) {
+      createPublicLink({ id: actionId });
+    } else {
+      openModal();
+    }
+  };
+
+  const handleDisablePublicLink = () => {
+    deletePublicLink({ id: actionId });
+  };
+
+  return (
+    <ActionSettingsContainer>
+      <SidebarContent title={t`Action settings`} onClose={onClose}>
+        <ActionSettingsContent>
+          <ToggleContainer>
+            <span>
+              <ToggleLabel htmlFor={id}>{t`Make public`}</ToggleLabel>
+              <Tooltip
+                tooltip={t`Creates a publicly shareable link to this action.`}
+              >
+                <Icon name="info" size={10} />
+              </Tooltip>
+            </span>
+            <Toggle id={id} value={isPublic} onChange={handleTogglePublic} />
```

Comment:
I deliberately use a different way of displaying the disable confirmation than in sharing questions and dashboards because they are rendering a different node so the animation on the toggle doesn't work.

https://github.com/metabase/metabase/blob/40bfc4a5543ba3f0fa1a23b97a4b0f6b5941ace9/frontend/src/metabase/public/components/widgets/SharingPane.tsx#L73-L99

#### toggle question/dashboard sharing
![toggle question public static](https://user-images.githubusercontent.com/1937582/214679048-9b6022a0-dcea-4ba2-9430-8210a498e71d.gif)

#### toggle action sharing
![toggle action public animation](https://user-images.githubusercontent.com/1937582/214679014-232d947e-9026-4499-b7c1-849674a3a907.gif)